bodhi/server/consumers/masher.py

```diff
@@ -358,12 +358,23 @@ def run(self):
                               self, str(self.compose), len(self.compose.updates))
                 self.save_state(ComposeState.initializing)
                 self.work()
+                notifications.publish(
+                    topic='repo.mash.complete',
+                    msg=dict(path=self.path,
+                             id=self.id,
+                             agent=self.agent))
         except Exception as e:
             with self.db_factory() as session:
                 self.db = session
                 self.compose = Compose.from_dict(session, self._compose)
                 self.compose.error_message = unicode(e)
                 self.save_state(ComposeState.failed)
+                notifications.publish(
+                    topic='repo.mash.failed',
+                    msg=dict(path=self.path,
+                             id=self.id,
+                             agent=self.agent),
+                    force=True)
```

Comment:
Are these two messages related to waiting on repo signatures? If not, I'd prefer not to include this change in the patch to keep the patch more targeted. Also, there already is a message sent upon completion and it indicates success/failure, so is this redundant? Lastly, if we do keep this, should we keep using the word "mash" since it's composes now? We can't rename the existing messages without bumping Bodhi's X version, but I've files a [ticket](https://github.com/fedora-infra/bodhi/issues/2151) to do that later.

Reply:
So, those two messages can be dropped, but I'll need `self.path` as argument to the `repo.signature.wait` message.